Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,124 @@
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public final class Log {
+
+    public private(set) static var destinations = [LogDestination]()
+
+    public static let defaultLevel = Level.error
+
+    // MARK:- Destination Management
+
+    public class func register(_ destination: LogDestination) {
+        if destinations.contains(where: { $0.instanceId == destination.instanceId }) == false {
+            destinations.append(destination)
+        }
+    }
+
+    public class func unregister(_ destination: LogDestination) {
+        destinations = destinations.filter({ registeredDestinaton -> Bool in
+            return registeredDestinaton.instanceId != destination.instanceId
+        })
+    }
+
+    public class func removeAllDestinations() {
+        destinations.removeAll()
+    }
+
+    // MARK:- Logging
+
+    public class func verbose(_ message: @autoclosure () -> String,
+                              file: String = #file,
+                              function: String = #function,
+                              line: Int = #line) {
+
+        log(level: .verbose, message: message, file: file, function: function, line: line)
+    }
+
+    public class func debug(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line) {
+
+        log(level: .debug, message: message, file: file, function: function, line: line)
+    }
+
+    public class func info(_ message: @autoclosure () -> String,
+                           file: String = #file,
+                           function: String = #function,
+                           line: Int = #line) {
+
+        log(level: .info, message: message, file: file, function: function, line: line)
+    }
+
+    public class func warning(_ message: @autoclosure () -> String,
+                              file: String = #file,
+                              function: String = #function,
+                              line: Int = #line) {
+
+        log(level: .warning, message: message, file: file, function: function, line: line)
+    }
+
+    public class func error(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line) {
+
+        log(level: .error, message: message, file: file, function: function, line: line)
+    }
+
+    public class func log(level: Level,
+                          message: @autoclosure () -> String,
+                          file: String = #file,
+                          function: String = #function,
+                          line: Int = #line) {
+
+        let item = Log.Item(level: level, message: message(), file: file,
+                            thread: threadName(), function: function, line: line)
+
+        for destination in destinations {
```

Comment:
To be sure that no destination blocks the calling thread, perhaps this should also be processed asynchronously.